### ISSUE TITLE
Mark all articles without keyword as "read" (inverse filter?)

### ISSUE THREAD

**Issue by Fogest** [FreshRSS/FreshRSS#3122]
Hey, I noticed you have a filter tool which can "Mark articles as read..." which contain a filter search. Is there any current way to do the inverse of this and have any articles *not* containing specified searches to be marked as read?

I'll provide a use case in case it helps make it more clear:

I often add RSS feeds from deal websites. I often don't care to see the majority of those noisey feeds in my main stream. I'd only like to see things that match filters I specify. For example if I am following a subreddit with PC part deals I may only be interested in monitors and GPU's at the current time. It would be nice to be able to put "monitor" and "GPU" each on a line and have it mark everything else as "read". 

Technically I could setup a user query, but this means I have to choose the visibility to hide from the main stream and have a separate user query I am checking which is not ideal.

**Comment by aledeg** [comment]
You could create an RSS output to your search matching your keywords. Then you can subscribe to your own feed and use that feed in your main view.

**Comment by Alkarex** [comment]
Yes, we implement the *not* operator, which should do the job.
Something like `!intitle:Alice !intitle:Bob`

**Comment by Fogest** [comment]
@aledeg That is good to know, I will go with @Alkarex solution for now but I may use that in the future for other things. A bit harder to modify what you're filtering on the fly with this approach.

@Alkarex If I was trying to only keep `monitor` and `gpu` as unread would I have to put it all in one line like you showed, or should it be on two lines? Also if this is case sensitive is there anyway to make it not case sensitive?

**Comment by Alkarex** [comment]
Only a single line with `-monitor -GPU`
(`!` or `-` as you want, they are synonym)
Try it first in your search field.

P.S.: Ah sorry, if you want to:
* Keep *monitor* **AND** *GPU*: `-monitor OR -GPU`
* Keep *monitor* **OR** *GPU*: `-monitor -GPU`

**Comment by Alkarex** [comment]
P.P.S.: Edited, I had my boolean logic confused this evening :-D

**Comment by Fogest** [comment]
Hmm now I am bit confused. If I use `-monitor -GPU` in the search now it is showing me only resulting that do not include "monitor" or "gpu" in them. If I am putting it in the filter would I use:

`!intitle:monitor OR !intitle:GPU`

Or am I now also getting my logic confused?

**Comment by Alkarex** [comment]
In the main search, it will show you matching articles. Those are the articles, which would be marked as read, if you use the same expression as an automatic filter.

Let me rephrase my previous post:

Use the following automatic filter if you want to automatically mark as read all articles except:
* articles which contain *monitor* as well as *GPU*: `-monitor OR -GPU`
* articles which contain either *monitor* or *GPU*: `-monitor -GPU`

The `intitle:XXX` prefix limits the search to the article's title, so `!intitle:XXX` searches only in the title, while `!XXX` searches in the whole article.

**Comment by Fogest** [comment]
Thank you, I think I got it now 👍
